presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analyzer.java

```diff
@@ -107,4 +109,12 @@ static void verifyNoAggregateWindowOrGroupingFunctions(Map<NodeRef<FunctionCall>
             throw new SemanticException(CANNOT_HAVE_AGGREGATIONS_WINDOWS_OR_GROUPING, predicate, "%s cannot contain aggregations, window functions or grouping operations: %s", clause, found);
         }
     }
+
+    static void verifyNoExternalFunctions(Map<NodeRef<FunctionCall>, FunctionHandle> functionHandles, FunctionManager functionManager, Expression predicate, String clause)
+    {
+        List<FunctionCall> externalFunctions = extractExternalFunctions(functionHandles, ImmutableList.of(predicate), functionManager);
+        if (!externalFunctions.isEmpty()) {
+            throw new SemanticException(NOT_SUPPORTED, predicate, "External functions in %s is not supported: %s", clause, externalFunctions);
```

Comment:
nit, for better readability of the constructed error message:
```
External functions in [%s] is not supported: %s
```

Reply:
There is no unit test coverage for this commit. Can we add some.

Reply:
> nit, for better readability of the constructed error message:
> 
> ```
> External functions in [%s] is not supported: %s
> ```

I don't think the [] is useful. It would make the error message like this:
```
External functions in [Lambda expression] is not supported:....
```